src/kvstore.c

```diff
@@ -568,14 +577,23 @@ kvstoreIterator *kvstoreIteratorInit(kvstore *kvs) {
 void kvstoreIteratorRelease(kvstoreIterator *kvs_it) {
     dictIterator *iter = &kvs_it->di;
     dictResetIterator(iter);
-
+    freeDictIfNeeded(kvs_it->kvs, kvs_it->didx);
```

Comment:
i'm not sure if I need to check `(kvs_it->didx != -1 && kvstoreGetDict(kvs_it->kvs, kvs_it->didx))` here

Reply:
I checked and it seems that it is not needed (i also wrote a test case to verify), but I still throw it out to seek a double check.